### PR TITLE
Include takendown posts for admins (feature branch)

### PR DESCRIPTION
**Pull request by devinivy**
Builds off of #1301, also looking ahead to #1233 and unifying the pds a bit with the bsky appview implementation.

### CODE REVIEW

File: packages/pds/src/app-view/api/app/bsky/feed/getAuthorFeed.ts [hunk 1]
```diff
@@ -26,19 +34,21 @@ export default function (server: Server, ctx: AppContext) {
...
-        throw new InvalidRequestError(
-          `Requester has blocked actor: ${actor}`,
-          'BlockedActor',
-        )
-      } else if (blocks.blockedBy) {
-        throw new InvalidRequestError(
-          `Requester is blocked by actor: $${actor}`,
-          'BlockedByActor',
-        )
+      if (requester) {
```
Comment (dholms) [inline packages/pds/src/app-view/api/app/bsky/feed/getAuthorFeed.ts:37]: a part of me just wants to two separate functions for "user getAuthorFeed" & "admin getAuthFeed" - not necessarily two different routes, just two functions that can call based on auth 🤔 

but also think this is totally fine & not that confusing. just depends on how much of a pattern we see this being going forward

Comment (dholms) [reply packages/pds/src/app-view/api/app/bsky/feed/getAuthorFeed.ts:37]: basically just that functions with a bunch of logic inside if statements are kinda hard to grok

**Review by dholms — APPROVED** [review]
look good 👍 

i'd just like to change around the requester parsing in getAuthorFeed a bit

File: packages/pds/src/app-view/api/app/bsky/feed/getAuthorFeed.ts (hunk 1 above)
Comment (devinivy) [reply packages/pds/src/app-view/api/app/bsky/feed/getAuthorFeed.ts]: Yupyup!  I took a quick pass at this to extract some logic to help with readability, and then combined the two conditional blocks for access-based auth rules into a single block.

File: packages/pds/src/app-view/api/app/bsky/feed/getAuthorFeed.ts [hunk 2]
```diff
@@ -1,19 +1,27 @@
...
 import { FeedRow } from '../../../../services/feed'
-import { InvalidRequestError } from '@atproto/xrpc-server'
 
 export default function (server: Server, ctx: AppContext) {
   server.app.bsky.feed.getAuthorFeed({
-    auth: ctx.accessVerifier,
+    auth: ctx.accessOrRoleVerifier,
     handler: async ({ req, params, auth }) => {
-      const requester = auth.credentials.did
+      const requester = 'did' in auth.credentials ? auth.credentials.did : null
```
[the remark this answers is not in the document]
Comment (devinivy) [reply packages/pds/src/app-view/api/app/bsky/feed/getAuthorFeed.ts]: Got a `type` discriminator in there 👍 feels much better.

**Pull request merged**